dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/tlvformat/FileTLVBlockReader.kt

```diff
@@ -0,0 +1,126 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.persistence.tlvformat
+
+import androidx.annotation.WorkerThread
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.internal.persistence.datastore.ext.toInt
+import com.datadog.android.core.internal.persistence.datastore.ext.toShort
+import com.datadog.android.core.internal.persistence.file.FileReaderWriter
+import com.datadog.android.core.internal.persistence.file.existsSafe
+import com.datadog.android.core.internal.persistence.file.lengthSafe
+import java.io.File
+import java.io.IOException
+import java.io.InputStream
+import java.util.Locale
+import java.util.concurrent.atomic.AtomicBoolean
+
+internal class FileTLVBlockReader(
+    val internalLogger: InternalLogger,
+    val fileReaderWriter: FileReaderWriter
+) {
+    private var endOfStream = AtomicBoolean(false)
+
+    @WorkerThread
+    internal fun all(file: File): List<TLVBlock> {
+        if (!file.existsSafe(internalLogger) || file.lengthSafe(internalLogger) == 0L) {
+            return arrayListOf()
+        }
+
+        val blocks = mutableListOf<TLVBlock>()
+        val stream = fileReaderWriter.readData(file).inputStream()
```

Comment:
if you open stream, you need also to close it. Also you probably want to have it buffered in order to reduce the number of syscalls. You can check some code here https://github.com/DataDog/dd-sdk-android/blob/a7837387c6a137beb317d49c8cdc66835c62aed4/dd-sdk-android-core/src/main/kotlin/com/datadog/android/core/internal/persistence/file/batch/PlainBatchFileReaderWriter.kt#L126-L141